### ISSUE TITLE
[Android] RNGestureHandlerEnabledRootView.java line 39, java.lang.IllegalStateException: Page can only be offset by a positive amount, not by -47

### ISSUE THREAD

**Issue by thanakij** [software-mansion/react-native-gesture-handler#1476]
## Description

We just receive a report from Crashlytics about a crash on Android. It was for RNGestureHandlerEnabledRootView.java line 39, **java.lang.IllegalStateException: Page can only be offset by a positive amount, not by -47**

```
Fatal Exception: java.lang.IllegalStateException: Page can only be offset by a positive amount, not by -47
       at androidx.viewpager2.widget.ScrollEventAdapter.updateScrollEventValues(ScrollEventAdapter.java:280)
       at androidx.viewpager2.widget.ScrollEventAdapter.onScrolled(ScrollEventAdapter.java:178)
       at androidx.recyclerview.widget.RecyclerView.dispatchOnScrolled(RecyclerView.java:5173)
       at androidx.recyclerview.widget.RecyclerView.scrollByInternal(RecyclerView.java:1971)
       at androidx.recyclerview.widget.RecyclerView.onTouchEvent(RecyclerView.java:3391)
       at androidx.viewpager2.widget.ViewPager2$RecyclerViewImpl.onTouchEvent(ViewPager2.java:991)
       at android.view.View.dispatchTouchEvent(View.java:14376)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3857)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3535)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView.dispatchTouchEvent(RNGestureHandlerEnabledRootView.java:39)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3863)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:3551)
       at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:730)
       at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1922)
       at android.app.Activity.dispatchTouchEvent(Activity.java:4051)
       at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
       ...
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1075)
```

## Steps To Reproduce

I cannot reproduce this. This is from Crashlytics.

### Expected behavior

Not crash.

### Actual behavior

Crash in some cases (not common).

## Package versions

- React: 16.11
- React Native: 0.62.3
- React Native Gesture Handler: 1.10.3

**Comment by jakub-gonet** [comment]
This crash isn't caused by RNGH.

This is snippet from RNGestureHandlerEnabledRootView.java:
```java
  @Override
  public boolean dispatchTouchEvent(MotionEvent ev) {
    if (mGestureRootHelper != null && mGestureRootHelper.dispatchTouchEvent(ev)) {
      return true;
    }
    return super.dispatchTouchEvent(ev); // line 39
  }
```
As you can see we delegate handling of the touch to the default implementation.

This crash is probably related to `androidx.viewpager2.widget` which crashes.

**Comment by eliaslecomte** [comment]
I'm also having this issue and having trouble fixing it.
I've read: https://stackoverflow.com/questions/59660691/java-lang-illegalstateexception-page-can-only-be-offset-by-a-positive-amount and it looks like it's due to animation properties of the FlatList -> RecyclerView.

**Comment by eliaslecomte** [comment]
In my case it ended up being an issue in react-native-pager-view that was fixed by bumping the version to >= 5.3.0
